index.test.ts

```diff
@@ -0,0 +1,182 @@
+import * as assert from 'assert';
+import * as extend from 'extend';
+import * as proxyquire from 'proxyquire';
+
+// for types
+// Importing from './index' rather than '.' allows us to depend on
+// index.ts rather than index.d.ts
+import * as _gcpMetadata from './index';
+type GcpMetadata = typeof _gcpMetadata;
+type RetryRequest =
```

Comment:
Is this type distinct from the type of the `request` module. If not, just use that like you do in `index.ts`.